pywr-schema/src/nodes/core.rs

```diff
@@ -170,7 +332,49 @@ impl LinkNode {
         Ok(vec![idx])
     }
     pub fn add_to_model(&self, network: &mut pywr_core::network::Network) -> Result<(), SchemaError> {
-        network.add_link_node(self.meta.name.as_str(), None)?;
+        let node_name = self.meta.name.as_str();
+        let link = network.add_link_node(node_name, None)?;
+        // add soft constrained nodes and aggregated node
+        match (&self.soft_min, &self.soft_max) {
+            (Some(_), None) => {
+                // add L_min and aggregated node for L and L_min
+                let soft_min_node = network.add_link_node(node_name, Self::soft_min_node_sub_name())?;
+                network.add_aggregated_node(
+                    node_name,
+                    Self::aggregated_node_sub_name(),
+                    &[vec![link], vec![soft_min_node]],
+                    None,
+                )?;
+            }
+            (None, Some(_)) => {
+                // add L_max and aggregated node for L and L_max
+                let soft_max_node = network.add_link_node(node_name, Self::soft_max_node_sub_name())?;
+                network.add_aggregated_node(
+                    node_name,
+                    Self::aggregated_node_sub_name(),
+                    &[vec![link], vec![soft_max_node]],
+                    None,
+                )?;
+            }
+            (Some(_), Some(_)) => {
+                // add L_min and L_max, and aggregated node for L, L_min and L_max
+                let soft_min_node = network.add_link_node(node_name, Self::soft_min_node_sub_name())?;
+                let soft_max_node = network.add_link_node(node_name, Self::soft_max_node_sub_name())?;
+                network.add_aggregated_node(
+                    node_name,
+                    Self::aggregated_node_sub_name(),
+                    &[vec![link], vec![soft_min_node], vec![soft_max_node]],
+                    None,
+                )?;
+                network.add_aggregated_node(
+                    node_name,
+                    Self::aggregated_node_l_l_min_sub_name(),
+                    &[vec![link], vec![soft_min_node]],
+                    None,
+                )?;
+            }
+            _ => {}
```

Comment:
Minor thing for if this ever updated. Using `(None, None)` instead of the `_` would prevent this accidentally catching unused combinations in the future.